crates/rpc-types/src/transaction.rs

```diff
@@ -35,7 +35,11 @@ pub struct Transaction<T = OpTxEnvelope> {
 
 impl<T: OpTransaction + TransactionTrait> Transaction<T> {
     /// Converts a consensus `tx` with an additional context `tx_info` into an RPC [`Transaction`].
-    pub fn from_transaction(tx: Recovered<T>, tx_info: OpTransactionInfo) -> Self {
+    pub fn from_transaction<TxIn>(tx: Recovered<TxIn>, tx_info: OpTransactionInfo) -> Self
```

Comment:
isn't this the same as calling this with `tx.convert()`?

Reply:
It is. But more importantly, we're facing the same issue as mentioned in a comment of alloy-rs/alloy#2591